components/ai_chat/renderer/page_text_distilling.cc

```diff
@@ -136,9 +140,33 @@ void DistillPageText(
       base::UTF16ToUTF8(base::JoinString(text_node_contents, u" "));
 
   if (contents_text.empty()) {
-    std::move(callback).Run({});
+    blink::WebScriptSource source = blink::WebScriptSource(
+        blink::WebString::FromASCII("document.body.innerText"));
+
+    auto on_script_executed =
+        [](base::OnceCallback<void(const absl::optional<std::string>&)>
+               callback,
+           absl::optional<base::Value> value, base::TimeTicks start_time) {
+          if (value->is_string()) {
+            std::move(callback).Run(value->GetString());
+            return;
+          }
+
+          std::move(callback).Run({});
+        };
+
+    render_frame->GetWebFrame()->RequestExecuteScript(
+        isolated_world_id, base::make_span(&source, 1u),
+        blink::mojom::UserActivationOption::kDoNotActivate,
+        blink::mojom::EvaluationTiming::kAsynchronous,
+        blink::mojom::LoadEventBlockingOption::kDoNotBlock,
+        base::BindOnce(on_script_executed, std::move(callback)),
+        blink::BackForwardCacheAware::kAllow,
+        blink::mojom::WantResultOption::kWantResult,
+        blink::mojom::PromiseResultOption::kAwait);
```

Comment:
I was worried about the lack of checking whether the calling class gets destroyed (or having a weakptr check in this lambda function), but I think that's covered by the provided callback being bound with a weakptr.